### ISSUE TITLE
Release Mockito 2.0

### ISSUE THREAD

**Issue by szczepiq** [mockito/mockito#123]
Incompatible changes:
- [x] stop producing mockito-all #153
- [x] stop depending on hamcrest internally #154
- [x] use newer hamcrest #232
- [x] make the anyXxx and any(Xxx) matchers intuitive #134, #194
- ~~fix the site links~~
- [x] push cglib mockmaker to a separate jar #248
- [x] stop using ant for producing OSGi bundles. No more ant in the build #249
- [x] remove jars from source distribution #250
- ~~perhaps introduce is() matcher~~ #246
- ~~richer and smarter stubbing~~ #303
- ~~support java8 features for capturing arguments and matchers~~
- [x] make the JUnitRule and the runner verbose
- [x] ensure release notes can be neatly generated for the RC #582

Possibly compatible (and hence, could be released pre-2.0 if needed)
- [x] drop deprecated code
- [x] unincubate API
- ~~drop serialVersionUID~~

**Comment by bric3** [comment]
No I don't think it is necessary, however we can make a CGLIB mock maker, for people that may depend on it.

**Comment by raphw** [comment]
I am curious: Do you guys have a time-line for the 2.0 release? I am more than happy to support any code-generation releated issues. However, my vacation is approaching so I will be gone for a couple of weeks starting mid-August.

**Comment by szczepiq** [comment]
It must happen in 2015. I hope we are ready in the next few months. In
theory, there's not much left work :)

Cheers!

On Fri, Jul 10, 2015 at 5:31 AM, Rafael Winterhalter <
notifications@github.com> wrote:

> I am curious: Do you guys have a time-line for the 2.0 release? I am more
> than happy to support any code-generation releated issues. However, my
> vacation is approaching so I will be gone for a couple of weeks starting
> mid-August.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mockito/mockito/issues/123#issuecomment-120399539>.
>



-- 
Szczepan Faber
Founder mockito.org; Core dev gradle.org
tweets as @szczepiq; blogs at blog.mockito.org

**Comment by bric3** [comment]
I think mockito 2.0 is nearly in shape to be released, most work should go in polishing remaining bits of the public API. IMHO current byte buddy version is fine, and already provide subtitle but needed enhancements.

More specifically on generics I believe current support is OK, after all mockito is a TDD tool for a rapid feedback loop, both on the dev workstation and the ci bots. I believe current generics support in the deepstubs answer is enough. About the issue you mention the support class I wrote is probably incomplete but enough, and I agree parsing this metadata is hard. Anyway at the moment if any extended support should be provided on this front it should be provided via a mockmaker plugin.

**Comment by raphw** [comment]
Thanks for the info. I just released Byte Buddy 0.7 but I recommend you to not yet update. Generic types are tricky and I already found a corner case in the bridge method resolution that does not work. I'll try to make it stable until the mid of August, though.

**Comment by davidkarlsen** [comment]
Any guess on when 2.0 will be released?

**Comment by szczepiq** [comment]
It's been silent for past couple of months, everybody busy, etc. Since
yesterday, I'm focusing my spare time on Mockito. I will not stop until 2.0
is out. We cannot guarantee any dates but 2.0 must be released in 2015.

On Sat, Sep 19, 2015 at 8:40 AM, David J. M. Karlsen <
notifications@github.com> wrote:

> Any guess on when 2.0 will be released?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mockito/mockito/issues/123#issuecomment-141681244>.
>



-- 
Szczepan Faber
Founder mockito.org; Core dev gradle.org
tweets as @szczepiq; blogs at blog.mockito.org

**Comment by patspam** [comment]
Any updates on when we might see a 2.0 release?

**Comment by raphw** [comment]
2015 is soon over ;) At some point, we need to do a cut in features. I think everybody is quite budy right now. As always, the time before Christmas are busy month for consultant work.

**Comment by szczepiq** [comment]
Soon. This year for sure. Sorry I was distracted! We will probably deliver
2.0 with what we have, no new fancy features ;)

**Comment by skabashnyuk** [comment]
Can this task https://github.com/mockito/mockito/issues/300 be included in 2.0 release?

**Comment by ChristianSchwarz** [comment]
Feel free to add #374 to the todo list

**Comment by wenerme** [comment]
2.0  is coming so slow.

**Comment by szczepiq** [comment]
It's way too slow. I'm on it. Thank you guys for pushing.

**Comment by ChristianSchwarz** [comment]
@szczepiq How can we/the community support the mockito core team ?

**Comment by szczepiq** [comment]
> @szczepiq <https://github.com/szczepiq> How can we/the community support
> the mockito core team ?
>
You already are. You guys are great. Thank you for patience and pushing us
to deliver.

The lagging of beta is my fault. I've started the beta cycle and I should
have completed it. I'm back in the game, though.

Once 2.x final is out we need clarify how community can drive supporting
and maintaining Mockito.

Note that Mockito is continuously delivered so you can open PR to change
the version to "2.1.0", and that's it :)

Cheers!

**Comment by davidkarlsen** [comment]
@szczepiq I left a pull request for removing deprecated code which is an 2.0 milestone issue - can you pull it/looks good?

**Comment by szczepiq** [comment]
I haven't reviewed yet. THANKS!

**Comment by TimvdLippe** [comment]
Per Mockito developers discussion:

We want to target publishing the release on July 31st with a release candidate on June 30th.

There is one issue which must be resolved for that, which is #194. Other than that I just found 1 minor TODO [in the Mockito documentation](https://github.com/mockito/mockito/blob/dff5510cb3e97dae414e58ba85825ad01b273c72/src/main/java/org/mockito/Mockito.java#L1091). The other issues can be fixed in a later minor version.

**Comment by skabashnyuk** [comment]
@TimvdLippe  what about https://github.com/mockito/mockito/issues/300?

**Comment by TimvdLippe** [comment]
@skabashnyuk I think that we still want to do this, but is not a blocker for the release. We must do it when we have time, but it is okay in my opinion to release it in a later minor version. I have to be honest that I was not participating in the original discussion so I do not know the exact details. @bric3 is the man to talk to :)

**Comment by bric3** [comment]
@TimvdLippe @skabashnyuk Humm I think we should really focus energy on mockito 2, TestNG could come later and maybe as a separate module.

**Comment by skabashnyuk** [comment]
>TestNG could come later.
@bric3 No problem. But why this can't be done in parallel? Code is ready, looks like this is an infrastructure-deployment issue. Or you are planning to do something bigger than just deploy artifacts?

**Comment by TimvdLippe** [comment]
@skabashnyuk Currently we are focused on releasing Mockito 2.0. This leaves little to no time for other issues sadly. If you are able to configure the deployment, we are happy to accept your pull request! We can devote some of our time to the TestNG issue after we have released Mockito 2.0.

**Comment by szczepiq** [comment]
Some update.

We're not ready for the release but there's good momentum. Let's plan out
what else needs to be done (announcement information, really documentation
around why to upgrade and how, solid plan what's next after the release
candidate, etc.).

I'm confident the RC will be out in August. I plan to announce everywhere
that we plan RB in August. If we don't meet the date there will high degree
of shame ;) Thoughts?

On Sat, Jun 25, 2016 at 3:59 AM, Tim van der Lippe <notifications@github.com
> wrote:

> @skabashnyuk <https://github.com/skabashnyuk> Currently we are focused on
> releasing Mockito 2.0. This leaves little to no time for other issues
> sadly. If you are able to configure the deployment, we are happy to accept
> your pull request! We can devote some of our time to the TestNG issue after
> we have released Mockito 2.0.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mockito/mockito/issues/123#issuecomment-228532419>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AABgp8m2jG-9Rj7USJ1V2s6GJNHLzbdwks5qPQodgaJpZM4C_fdm>
> .
>



-- 
Szczepan Faber
Founder @ mockito.org | Twitter @ szczepiq
Author @ https://www.linkedin.com/today/author/6016386

**Comment by bric3** [comment]
I think most of the work should be focused on the upgrade guide, there's not much left for 2.0, follow up tasks can be completed in 2.1

**Comment by TimvdLippe** [comment]
I think we can publish the release candidate while working on the actual upgrade guide (state this in the github release for users to follow). The only two issues that are non-documentation are #384 with PR #527 of @szczepiq and #489 with PR #491 of myself. #527 has some feedback to be processed and #491 is more of a discussion regarding the runner.

Once these two are resolved, we can merge #483 and the release candidate is published :tada: 

For an up-to-date status, see https://waffle.io/mockito/mockito and https://github.com/mockito/mockito/milestone/1

**Comment by szczepiq** [comment]
This is great stuff!!! Thanks a lot Tim for pushing the release forward & organizing it so well.

I'm making good progress on the warnings stuff for JUnit rule and the listener - will update the PR very soon.

**Comment by szczepiq** [comment]
I'm confused why this ticket is closed. I don't see 2.0 in maven central :)

**Comment by TimvdLippe** [comment]
@szczepiq because this issue was referenced in #483 which I just merged

**Comment by szczepiq** [comment]
ah, GH smartness :)

**Comment by szczepiq** [comment]
As promised here’s suggested plan to get 2.0 out. Please give feedback, sing up for tasks and commit to ETA so that we get 2.0 out :)

Step 1:
 - [ ] prepare RC release notes + announcement (highlights, motivation, request for feedback) - #582
 - [ ] review and update documentation for 2.0.0 - #596
 - [ ] ensure & test manual Mockito releases - #586
 - [ ] update CI / build scripts (releasing betas from branch, etc) - #594

Step 2:
 - [ ] branch out / fixes to release / Travis - #594

Step 3:
 - [ ] pull the trigger for RC!!!!!, close darn #123 !!!!!!

**Comment by TimvdLippe** [comment]
![image](https://cloud.githubusercontent.com/assets/5948271/18092170/97fac1ca-6ecb-11e6-83e0-fd2953833359.png)

**Comment by TimvdLippe** [comment]
I think we have updated all relevant documentation. @bric3 once you branch out to `release/2.x` we can publish the release candidate :tada:

**Comment by TimvdLippe** [comment]
I went ahead and published the release candidate. You can download `2.1.0-RC.1` at https://bintray.com/szczepiq/maven/mockito/2.1.0-RC.1#

**Comment by bric3** [comment]
By the way I would prefer to have mockito account on bintray

**Comment by TimvdLippe** [comment]
Yes agreed. Let's extract a separate issue for that.

**Comment by TimvdLippe** [comment]
Mockito 2.1.0 has been released, upgrade today :tada:
